rust/src/csv/reader.rs

```diff
@@ -70,21 +204,46 @@ impl Reader {
     /// Create a new CsvReader
     pub fn new(
         file: File,
-        schema: Arc<Schema>,
+        schema: Option<Arc<Schema>>,
```

Comment:
With these many parameters, we should implement a `ReaderBuilder`.

Reply:
I implemented a `ReaderBuilder`. Please let me know if it's fine the way that I did it. I initially created builder methods that returned a `&mut ReaderBuilder` (e.g. pub fn with_delimiter(&self) -> &mut Self`, but using the methods in examples was PITA. So I used a builder that returns a new copy each time.